### PR TITLE
Switch to ORC for LLVM-based code generation.

### PR DESCRIPTION
**Pull request by vext01**
This initial iteration leaks the JITed code and stackmaps. To do the correct thing we would store the `LLJIT` object (freshly released from its `unique_ptr` so as to stop it dropping) and carry it around until we want to delete the underlying memory (probably inside `CompiledTrace::drop`).

(Note that even with that fix, all JITted code is still immortal, because we store code in a way that Drop won't run anyway)

This change also disables code generation for the `trace_compiler` suite. I was having a hard time getting it to link, but then realised that for what this suite is designed to do, no code generation is required. In turn some other hacks could go.

`GlobalMappings` appears to not be needed any more. ORC seems to be able to find symbols in the current address space automatically. It's a largeish change, so I think we should kill this in a follow-up PR.

For more information on ORC, see:
https://llvm.org/docs/ORCv2.html

### CODE REVIEW

**Comment by vext01** [comment]
I should check yklua still works with this change. Please wait until I have before merging.

File: yktracec/src/stackmap_oll_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// LLVM ORC Object Linking Layer plugin for preserving and locating stackmaps.
+
+#include "stackmap_oll_plugin.h"
+
+#ifdef __linux__
```
Comment (ltratt) [inline yktracec/src/stackmap_oll_plugin.cc:5]: Presumably we need an `#else #error unsupported platform` or similar, to stop "undefined variable SMSectionName" confusion?

**Comment by vext01** [comment]
> I should check yklua still works with this change

Our `test.sh` still works just fine.

File: yktracec/src/stackmap_oll_plugin.cc (hunk 1 above)
Comment (vext01) [reply yktracec/src/stackmap_oll_plugin.cc:5]: Fixed in 21d0acad

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**